### ISSUE TITLE
[BUG]: basePath is not working except default value

### ISSUE THREAD

**Issue by PerfectPan** [dai-shi/waku#698]
Reproduce Repo: https://github.com/PerfectPan/waku-issue-691

Setting basePath manualy in waku.config.ts will cause error except `/`.
```ts
/** @type {import('waku/config').Config} */
export default {
  basePath: '/test/'
};
```

build mode: load js and css file error(`pnpm build && pnpm start`)
![CleanShot 2024-05-05 at 14 01 46@2x](https://github.com/dai-shi/waku/assets/24316656/b68ec33e-cc78-4b85-9153-3a010cc1fc4f)

dev mode: unknown error(`pnpm dev`)
![CleanShot 2024-05-05 at 13 57 04@2x](https://github.com/dai-shi/waku/assets/24316656/7ee2cc14-a238-4031-9988-6a8d4e9bc128)

Some previous reserach can be seen in https://github.com/dai-shi/waku/pull/697#issuecomment-2094563046

Would like to solve it later and also welcome other one to work on this.

**Comment by dai-shi** [comment]
Thanks for opening an issue.
Let's fix it step by step:
1. SSG
2. build
3. dev

By SSG, I mean to build static files and serve from a different folder:
```bash
pnpm builld
mkdir ssg
cp -r dist/public ssg/test
npx serve -s ssg
```

**Comment by yuusheng** [comment]
Hi! I'm trying to solve this. But I encounter some problems:

1. It seems that we should not add basePath on `entriesFile` in devServerImpl.ts, as it will be a true file path when vite handle this path(It should be `.../waku_project/src/entries` instead of `.../waku_project/test/src/entries`)
2. But when I use `const entriesFile = ${vite.config.root}/${config.srcDir}/${SRC_ENTRIES}\;`, `globalThis.__WAKU_PREFETCHED__` whill no longer inject by the ssr middleware. So it will fetch `http://localhost:3000/RSC/test.txt`. I can't figure out how to solve this. Do you have some recommends @dai-shi ?

**Comment by dai-shi** [comment]
> 1. It seems that we should not add basePath to `entriesFile` in devServerImpl.ts

Do we do that? Maybe, it's for SSG, but not sure about DEV.

I haven't looked into this yet, so my question might be appropriate, but can you describe the issue you are tackling? Aren't there several issues for `basePath` to work as expected?

**Comment by yuusheng** [comment]
> Do we do that? Maybe, it's for SSG, but not sure about DEV.
Yes. I will take a look at how it works in SSG.

Reproduction repo: [https://github.com/yuusheng/waku](https://github.com/yuusheng/waku)
<img width="868" alt="image" src="https://github.com/user-attachments/assets/c97b1e40-8594-4b25-a4a3-c5f1fbac5d56">
I suppose this is caused by function `fetchRSC`. The global var should be injected `globalThis.__WAKU_PREFETCHED__` is missing.

**Comment by dai-shi** [comment]
@yuusheng Thanks for the investigation.
'prefetching' is an optional feature and, while we need to fix it, it should work without prefetching at first.

https://github.com/dai-shi/waku/blob/83bf5ee63aa7c9b5959afb0552afd34797995492/packages/waku/src/client.ts#L27-L29

Can you check if `BASE_PATH` is set appropriately?

**Comment by dai-shi** [comment]
Okay, there must be something I miss. So, `http://localhost:3004/test/RSC/test.txt` is 404. What's it serving then? `http://localhost:3004/RSC/test.txt`?

**Comment by yuusheng** [comment]
Nope. http://localhost:3004/RSC/test.txt is served by waku's router I think. the `content-type` is text/html
<img width="966" alt="image" src="https://github.com/user-attachments/assets/b65b566e-5367-49c7-8feb-be6dc30c4ca4">

**Comment by dai-shi** [comment]
> the `content-type` is text/html

wait, so is it an html content instead of an RSC payload?
